src/commands/reword.rs

```diff
@@ -52,7 +62,19 @@ pub fn reword(
         None => return Ok(1),
     };
 
-    let messages = match build_messages(&messages, &commits, get_comment_char(&repo)?)? {
+    let edit_message_fn = |message: &str| {
+        // Editor::edit() normally requires that the file being edited is saved before the editor
+        // is closed. If it's not, it will return None; and in all other cases, it will return
+        // Some(message). We don't care if the file has been saved, only if it hasn't changed, so
+        // here we call `require_save(false)` and just ignore the None case.
+        let result = Editor::new()
+            .require_save(false)
+            .edit(message)?
+            .expect("`Editor::edit` should not return `None` when `require_save` is `false`");
+        Ok(result)
+    };
```

Comment:
Sorry if I'm being dense, but what does this closure do for us vs just leaving it inline? I don't see it being used anywhere else, or being swapped out for another closure or something like that. Does it have something to do w/ the `#[instrument(skip(edit_message_fn))]` line?

Reply:
I just thought it would be hard to read the next line, since it's the subject of the `match` expression.